Library/ViewModels/DiscoveryPageViewModel.swift

```diff
@@ -351,6 +372,34 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
         )
       }.skipRepeats()
 
+    // MARK: Personalization Callout Card
+
+    let viewReady = Signal.combineLatest(
+      self.optimizelyClientConfiguredProperty.signal,
+      editorialHeaderShouldShow
+    )
+
+    self.showPersonalization = Signal.merge(
+      viewReady,
+      viewReady.takeWhen(self.onboardingCompletedProperty.signal)
+    )
+    .map(second)
+    .map { shouldShowHeader in
+      if shouldShowHeader {
+        return shouldShowPersonalization()
```

Comment:
What if we passed the `shouldShowHeader` boolean into the `shouldShowPersonalization()` function and then just had a single `return`?

Reply:
They're kinda separate things - `shouldShowPersonalization` deals purely with the personalization experiment logic, while `shouldShowHeader` is more generally about whether we're in a discovery category that allows headers (even editorial headers) to be shown.

Reply:
I could maybe simplify this to `return shouldShowHeader && shouldShowPersonalization()` though.